crates/walrus-service/src/node.rs

```diff
@@ -1167,9 +1172,16 @@ impl StorageNode {
             return Ok(false);
         };
 
-        if self.inner.current_event_epoch() >= first_complete_epoch {
+        let Some(current_event_epoch) = self.inner.try_get_current_event_epoch() else {
```

Comment:
Previously, when processing events without epoch before first epoch change start (e.g., CheckpointBoundary), current_event_epoch() is set to 0, so this is ok. Now, the logic is changed to whether `current_event_epoch` is initialized in the tokio watch channel.